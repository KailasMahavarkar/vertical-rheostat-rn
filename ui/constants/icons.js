const CDN_PATH = 'https://imgd.aeplcdn.com/';
const skyline = `${CDN_PATH}0x0/cw/static/icons/skyline-bg.png`;

export { skyline };
