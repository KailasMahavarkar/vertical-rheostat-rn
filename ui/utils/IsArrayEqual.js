export const isArrayEqual = (array1, array2) => {
    if (!array1 || !array2) return false;
    if (array1.length != array2.length) return false;
    for (let i = 0, l = array1.length; i < l; i++) {
        if (array1[i] instanceof Array && array2[i] instanceof Array) {
            if (!array1[i].equals(array2[i])) return false;
        } else if (array1[i] != array2[i]) {
            return false;
        }
    }
    return true;
};
