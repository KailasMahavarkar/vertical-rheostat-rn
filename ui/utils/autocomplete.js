import escapeStringRegexp from 'escape-string-regexp';

const CASE_INSENSITIVE = 'i';
const COMBINING_MARKS = /[\u0300-\u036F]/;

export const getMatchBounds = (subject, str) => {
    const search = new RegExp(escapeStringRegexp(str), CASE_INSENSITIVE);

    const matches = search.exec(subject);

    if (!matches) {
        return null;
    }

    let start = matches.index;
    let matchLength = matches[0].length;

    // Account for combining marks, which changes the indices.
    if (COMBINING_MARKS.test(subject)) {
        // Starting at the beginning of the subject string, check for the number of
        // combining marks and increment the start index whenever one is found.
        for (let ii = 0; ii <= start; ii++) {
            if (COMBINING_MARKS.test(subject[ii])) {
                start += 1;
            }
        }

        // Similarly, increment the length of the match string if it contains a
        // combining mark.
        for (let ii = start; ii <= start + matchLength; ii++) {
            if (COMBINING_MARKS.test(subject[ii])) {
                matchLength += 1;
            }
        }
    }

    return {
        end: start + matchLength,
        start,
    };
};

export const getOptionLabel = (option, labelKey) => {
    if (option.paginationOption || option.customOption) {
        return option[typeof labelKey === 'string' ? labelKey : 'label'];
    }

    let optionLabel;

    if (typeof option === 'string') {
        optionLabel = option;
    }
    if (typeof labelKey === 'function') {
        // This overwrites string options, but we assume the consumer wants to do
        // something custom if `labelKey` is a function.
        optionLabel = labelKey(option);
    } else if (
        typeof labelKey === 'string' &&
        typeof option == 'object' &&
        Object.getPrototypeOf(option) == Object.prototype
    ) {
        optionLabel = option[labelKey];
    }

    return optionLabel;
};
