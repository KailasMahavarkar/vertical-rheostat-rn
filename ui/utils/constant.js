export const ORIENTATION = {
    horizontal: 'horizontal',
    vertical: 'vertical',
};

export const COLOR = {
    // neutral
    black: 'neutral-1200',
    charcoal: 'neutral-1000',
    dimGray: 'neutral-900',
    darkGray: 'neutral-800',
    silver: 'neutral-700',
    lightGray: 'neutral-600',
    gainsBoroGray: 'neutral-500',
    whiteSmoke: 'neutral-400',
    whiteSmokeLight: 'neutral-200',
    dimWhite: 'neutral-300',
    white: 'neutral-100',
    // info
    'brown-darkest': 'info-1100',
    'brown-dark': 'info-1000',
    mediumBrown: 'info-900',
    brown: 'info-700',
    'brown-light': 'info-500',
    offBrown: 'info-300',
    // accent
    'blue-darkest': 'accent-1100',
    'blue-dark': 'accent-1000',
    blue: 'accent-900',
    'blue-light': 'accent-800',
    'blue-lightest': 'accent-700',
    // secoondary
    'teal-dark': 'secondary-1000',
    teal: 'secondary-900',
    'teal-light': 'secondary-700',
    'teal-lightest': 'secondary-400',
    aquaTeal: 'secondary-300',
    // primary
    'red-darkest': 'primary-1100',
    'red-dark': 'primary-1000',
    red: 'primary-900',
    'red-light': 'primary-700',
    'red-lightest': 'primary-400',
    redRose: 'primary-300',
    // misc
    blackRussian: 'misc-500',
    edenBlue: 'misc-400',
    gothic: 'misc-300',
    brightTurquoise: 'misc-200',
    transparent: 'misc-10',
};
