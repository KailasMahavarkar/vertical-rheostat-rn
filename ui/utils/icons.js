// Need to remove once icon component is built

const cdnPath = 'https://imgd.aeplcdn.com';
export const NO_CAR_IMAGE = `${cdnPath}/0x0/no-image/no-image.png`;
