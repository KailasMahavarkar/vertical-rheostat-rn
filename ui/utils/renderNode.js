import React from 'react';
import Text from '../Text';

export const renderNode = (element, props) => {
    return typeof element === 'string' ? (
        <Text {...props}>{element}</Text>
    ) : (
        element
    );
};
