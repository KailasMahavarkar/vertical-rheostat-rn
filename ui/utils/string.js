export const concatString = (prefix, suffix = '', infix = '-') => {
    if (prefix) {
        const concatedString = `${prefix} ${suffix}`;
        return concatedString.trim().replace(/\s+/g, infix);
    }
    return null;
};
