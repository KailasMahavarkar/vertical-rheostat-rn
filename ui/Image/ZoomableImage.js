import React, { Component } from 'react';
import { View, PanResponder, Image, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { NO_CAR_IMAGE } from '../utils/icons';
import { concatString } from '../utils/string';
import { testIdProps } from '../utils/testIdProps';

const propTypes = {
    /* Image width */
    width: PropTypes.number.isRequired,
    /** Aspect ratio of image */
    aspectRatio: PropTypes.oneOf(['21:9', '16:9', '4:3', '1:1']),
    /* Value for data-testing-id attribute */
    'data-testing-id': PropTypes.string.isRequired,
    /* Doubletap delay */
    doubleTapDelay: PropTypes.number,
    /* Maximum scale value for zoom feature */
    maxScale: PropTypes.number,
    /** Callback fire when mouse wheel event triggers. */
    onDoubleTap: PropTypes.func,
    /** Callback fire when pinch event triggers. */
    onPinch: PropTypes.func,
    /** Pinch provision enabled if true */
    shouldZoomOnPinch: PropTypes.bool,
    /* Callback fired when touchmove event triggers. */
    onTouchMove: PropTypes.func,
    /** Alternate Image link if the image doesn't exist */
    placeholderSrc: PropTypes.string,
    /** Default image if the image takes time to load */
    defaultSrc: PropTypes.any,
    /** Image url */
    src: PropTypes.string,
    /* Custom styles for image */
    style: PropTypes.oneOfType(PropTypes.object, PropTypes.array),
    /* Threshold scale value for first double tap functionality */
    threshold: PropTypes.number,
    /* Custom styles for image wrapper */
    wrapperStyle: PropTypes.oneOfType(PropTypes.object, PropTypes.array),
    /* Threshold swipe value for swiping functionality */
    thresholdSwipeValue: PropTypes.number,
};

const defaultProps = {
    width: 360,
    aspectRatio: '16:9',
    doubleTapDelay: 500,
    maxScale: 3,
    onDoubleTap: () => {},
    shouldZoomOnPinch: false,
    thresholdSwipeValue: 50,
    onPinch: () => {},
    onTouchMove: () => {},
    onTouchRelease: () => {},
    placeholderSrc: NO_CAR_IMAGE,
    defaultSrc: NO_CAR_IMAGE,
    src: '',
    style: null,
    threshold: 1.5,
    wrapperStyle: null,
};

const ASPECT_RATIO = {
    '21:9': 21 / 9,
    '16:9': 16 / 9,
    '4:3': 4 / 3,
    '1:1': 1,
};

const styles = StyleSheet.create({
    wrapperStyle: {
        overflow: 'hidden',
    },
    imageStyle: {
        position: 'absolute',
    },
});

class ZoomableImage extends Component {
    constructor(props) {
        super(props);

        this.state = {
            scale: 1,
            initialX: null,
            initalY: null,
            top: 0,
            left: 0,
            hasSrc: props.src.length > 0,
            lastTap: 0,
            singlePressTimer: 0,
        };

        this.zoomGlobalStates = {
            isMoving: false,
            isZooming: false,
            initialDistance: null,
            initialZoom: 1,
            initialTopWithoutZoom: 0,
            initialLeftWithoutZoom: 0,
            initialTop: 0,
            initialLeft: 0,
            gestureType: null,
            gestureDirection: null,
        };

        this.height = this.getImageHeight(props.width);
    }

    componentDidUpdate(prevProps, prevState) {
        const { resetScale, width } = this.props;
        if (prevProps.resetScale !== resetScale) {
            this.setState({ scale: 1, top: 0, left: 0 });
        }

        if (prevProps.width !== width) {
            this.height = this.getImageHeight(width);
        }
    }

    getImageHeight = width => {
        const { aspectRatio } = this.props;
        return (1 / ASPECT_RATIO[aspectRatio]) * width;
    };

    maxOffset = (offset, imageDimension) => {
        const { scale } = this.state;
        const max = imageDimension - imageDimension * scale;
        if (max >= 0) {
            return 0;
        }
        return offset < max ? max : offset;
    };

    calcOffsetByZoom = () => {
        const { width } = this.props;
        const { scale } = this.state;
        const height = this.getImageHeight(width);

        const xDiff = width * scale - width;
        const yDiff = height * scale - height;
        return {
            left: -xDiff / 2,
            top: -yDiff / 2,
        };
    };

    handlePinch(x1, y1, x2, y2) {
        const dx = Math.abs(x1 - x2);
        const dy = Math.abs(y1 - y2);
        const distance = Math.sqrt(Math.pow(dx, 2) + Math.pow(dy, 2));

        const center = {
            x: x1 > x2 ? x1 - (x1 - x2) / 2 : x2 - (x2 - x1) / 2,
            y: y1 > y2 ? y1 - (y1 - y2) / 2 : y2 - (y2 - y1) / 2,
        };

        const { scale, top, left } = this.state;
        const { width, maxScale } = this.props;
        const height = this.height;
        const {
            isZooming,
            initialDistance,
            initialZoom,
            initialTopWithoutZoom,
            initialLeftWithoutZoom,
        } = this.zoomGlobalStates;

        if (!isZooming) {
            const offsetByZoom = this.calcOffsetByZoom();
            this.setState({
                initialX: center.x,
                initialY: center.y,
            });
            this.zoomGlobalStates = {
                ...this.zoomGlobalStates,
                isZooming: true,
                initialDistance: distance,
                initialZoom: scale,
                initialTopWithoutZoom: top - offsetByZoom.top,
                initialLeftWithoutZoom: left - offsetByZoom.left,
            };
        } else {
            const touchZoom = distance / initialDistance;
            const scale =
                touchZoom * initialZoom > 1 ? touchZoom * initialZoom : 1;

            const offsetByZoom = this.calcOffsetByZoom();
            const left = initialLeftWithoutZoom * touchZoom + offsetByZoom.left;
            const top = initialTopWithoutZoom * touchZoom + offsetByZoom.top;

            if (scale < maxScale + 0.1) {
                this.setState({
                    scale,
                    left: left > 0 ? 0 : this.maxOffset(left, width),
                    top: top > 0 ? 0 : this.maxOffset(top, height),
                });
            }
        }
    }

    handleTouchMove(event, x, y, gestureState) {
        const { top, left, initialX, initialY, scale } = this.state;
        const { width, thresholdSwipeValue, onTouchMove } = this.props;
        const height = this.height;
        const { isMoving, initialTop, initialLeft } = this.zoomGlobalStates;
        const { dx } = gestureState;
        if (dx < 0 && -1 * dx > thresholdSwipeValue) {
            this.zoomGlobalStates.gestureDirection = 'left';
        } else if (dx > thresholdSwipeValue) {
            this.zoomGlobalStates.gestureDirection = 'right';
        } else {
            this.zoomGlobalStates.gestureDirection = null;
        }
        onTouchMove(event, scale, this.zoomGlobalStates.gestureDirection);
        if (!isMoving) {
            this.setState({
                initialX: x,
                initialY: y,
            });
            this.zoomGlobalStates = {
                ...this.zoomGlobalStates,
                isMoving: true,
                initialTop: top,
                initialLeft: left,
            };
        } else {
            const left = initialLeft + x - initialX;
            const top = initialTop + y - initialY;

            this.setState({
                left: left > 0 ? 0 : this.maxOffset(left, width),
                top: top > 0 ? 0 : this.maxOffset(top, height),
            });
        }
    }

    handlePanResponderMove = (event, gestureState) => {
        const { onPinch, onTouchMove, shouldZoomOnPinch } = this.props;
        const { scale } = this.state;
        const touches = event.nativeEvent.touches;
        const { isZooming } = this.zoomGlobalStates;
        const { vx, vy } = gestureState;
        if (Math.abs(vx) > 0 || Math.abs(vy) > 0) {
            if (touches.length === 2 && shouldZoomOnPinch) {
                this.zoomGlobalStates.gestureType = 'pinch';
                onPinch(event, scale);
                this.handlePinch(
                    touches[0].pageX,
                    touches[0].pageY,
                    touches[1].pageX,
                    touches[1].pageY
                );
            } else if (touches.length === 1 && !isZooming) {
                this.handleTouchMove(
                    event,
                    touches[0].pageX,
                    touches[0].pageY,
                    gestureState
                );
                this.zoomGlobalStates.gestureType = 'shift';
                onTouchMove(event);
            }
        }
    };

    getNextPositionValues = (nextScale, locationX, locationY) => {
        const { top, left, scale } = this.state;
        const incrementalScalePercentage = (nextScale - scale) / scale;
        const translateY = locationY * incrementalScalePercentage;
        const translateX = locationX * incrementalScalePercentage;
        const nextTop = top - translateY;
        const nextLeft = left - translateX;

        return { nextLeft, nextTop };
    };

    handleDoubleTap = e => {
        const { threshold, maxScale, onDoubleTap } = this.props;
        const { scale } = this.state;
        const { locationY, locationX } = e.nativeEvent;

        if (scale < threshold) {
            onDoubleTap(e, threshold);
            const { nextLeft, nextTop } = this.getNextPositionValues(
                threshold,
                locationX,
                locationY
            );

            this.setState({
                scale: threshold,
                left: nextLeft,
                top: nextTop,
            });
        } else if (scale >= threshold && scale < maxScale) {
            onDoubleTap(e, maxScale);
            const { nextLeft, nextTop } = this.getNextPositionValues(
                maxScale,
                locationX,
                locationY
            );

            this.setState({
                scale: maxScale,
                left: nextLeft,
                top: nextTop,
            });
        } else if (scale >= maxScale) {
            this.setState({ scale: 1, left: 0, top: 0 }, () => {
                onDoubleTap(e, 1);
            });
        }
    };

    handleSingleTap = e => {
        this.cancelSinglePressTimer();
        const { doubleTapDelay, onSingleTap } = this.props;
        const { lastTap } = this.state;
        const now = Date.now();
        if (lastTap && now - lastTap < doubleTapDelay) {
            this.handleDoubleTap(e);
        } else {
            this.setState({ lastTap: now });
            const timeout = setTimeout(() => {
                this.setState({ lastTap: 0 });
                if (onSingleTap) {
                    onSingleTap(e);
                }
            }, doubleTapDelay);
            this.setState({ singlePressTimer: timeout });
        }
    };

    cancelSinglePressTimer = () => {
        const { singlePressTimer } = this.state;
        if (singlePressTimer) {
            clearTimeout(singlePressTimer);
            this.setState({ singlePressTimer: 0 });
        }
    };
    handlePanResponderRelease = event => {
        const { onTouchRelease } = this.props;
        const { scale } = this.state;

        this.zoomGlobalStates.gestureType == 'shift' &&
            onTouchRelease &&
            onTouchRelease(
                event,
                scale,
                this.zoomGlobalStates.gestureDirection
            );
        if (!this.zoomGlobalStates.gestureType) {
            this.handleSingleTap(event);
        }
        this.zoomGlobalStates = {
            ...this.zoomGlobalStates,
            isMoving: false,
            isZooming: false,
            gestureType: null,
        };
    };

    componentWillMount() {
        this._panResponder = PanResponder.create({
            onStartShouldSetPanResponder: () => true,
            onPanResponderMove: this.handlePanResponderMove,
            onPanResponderRelease: this.handlePanResponderRelease,
            onShouldBlockNativeResponder: () => true,
        });
    }

    handleOnError = () => {
        this.setState({ hasSrc: false });
    };

    render() {
        const {
            wrapperStyle,
            style,
            src,
            placeholderSrc,
            width,
            'data-testing-id': testingName,
            defaultSrc,
        } = this.props;
        const { top, hasSrc, left, scale } = this.state;
        const height = this.height;
        const imageSrc = hasSrc ? src : placeholderSrc;

        const imgWrapperStyle = [
            styles.wrapperStyle,
            wrapperStyle,
            {
                width,
                height,
            },
        ];

        const imgStyle = [
            styles.imageStyle,
            style,
            {
                top,
                left,
                width: width * scale,
                height: height * scale,
            },
        ];

        return (
            <View
                style={imgWrapperStyle}
                {...this._panResponder.panHandlers}
                {...testIdProps(
                    concatString(testingName, 'zoomable-image-wrapper')
                )}
            >
                <Image
                    style={imgStyle}
                    source={{ uri: imageSrc }}
                    onError={this.handleOnError}
                    defaultSource={defaultSrc}
                    {...testIdProps(
                        concatString(testingName, 'zoomable-image')
                    )}
                />
            </View>
        );
    }
}

ZoomableImage.propTypes = propTypes;
ZoomableImage.defaultProps = defaultProps;

export default ZoomableImage;
