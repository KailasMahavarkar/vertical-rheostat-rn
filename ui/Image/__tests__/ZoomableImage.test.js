import React from 'react';
import { Image, View } from 'react-native';
import { create } from 'react-test-renderer';

import ZoomableImage from '../ZoomableImage';

const mockFn = jest.fn();
const defaulImageSrc = require('../assets/no_cars_img.png');
const allProps = {
    width: 360,
    aspectRatio: '21:9',
    threshold: 2,
    maxScale: 4,
    src:
        'https://imgd.aeplcdn.com/0x0/n/cw/ec/41523/sonet-exterior-right-front-three-quarter-108.jpeg?q=85',
    placeholderSrc:
        'https://imgd.aeplcdn.com/310x174/n/cw/ec/54437/2021-wrangler-exterior-right-front-three-quarter.jpeg?q=75',
    defaultSrc: defaulImageSrc,
    onTouchMove: mockFn,
    onPinch: mockFn,
    onDoubleTap: mockFn,
    wrapperStyle: { padding: 10 },
    style: { padding: 10 },
    doubleTapDelay: 300,
    'data-testing-id': 'parent',
};

const findAllByTestID = (instance, testId, component) =>
    instance.root.findAll(
        el => el.props.testID === testId && el.type === component
    );

import { NO_CAR_IMAGE } from '../../utils/icons';

describe('<ZoomableImage />', function() {
    describe('render', function() {
        it('Should render with default props', function() {
            const wrapper = mount(<ZoomableImage width={allProps.width} />);

            expect(wrapper).toMatchSnapshot();
        });

        it('Should render with custom props', function() {
            const wrapper = mount(<ZoomableImage {...allProps} />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('props', function() {
        describe('width', function() {
            it('Should add given width to Image component', function() {
                const wrapper = mount(<ZoomableImage width={allProps.width} />);

                const image = wrapper.find(Image);
                const width = image.props().style[2].width;

                expect(width).toBe(allProps.width);
            });
        });

        describe('aspectRatio', function() {
            it('Should apply appropriate height with default aspectRatio i.e. 16:9', function() {
                const wrapper = mount(<ZoomableImage width={allProps.width} />);

                const image = wrapper.find(Image);
                const height = image.props().style[2].height;

                expect(height).toBe((allProps.width * 9) / 16);
            });

            it('Should apply same height as width when aspectRatio is 1:1', function() {
                const wrapper = mount(
                    <ZoomableImage width={allProps.width} aspectRatio="1:1" />
                );

                const image = wrapper.find(Image);
                const height = image.props().style[2].height;

                expect(height).toBe(allProps.width);
            });
        });

        describe('data-testing-id', function() {
            it('Should apply null testID with blank data-testing-id', function() {
                const wrapper = create(
                    <ZoomableImage width={allProps.width} />
                );

                const viewCount = findAllByTestID(
                    wrapper,
                    'zoomable-image-wrapper',
                    'View'
                ).length;

                expect(viewCount).toEqual(0);
            });

            it('Should apply given testID with data-testing-id', function() {
                const wrapper = create(
                    <ZoomableImage
                        width={allProps.width}
                        data-testing-id={allProps['data-testing-id']}
                    />
                );

                const viewCount = findAllByTestID(
                    wrapper,
                    `${allProps['data-testing-id']}-zoomable-image-wrapper`,
                    'View'
                ).length;

                expect(viewCount).toEqual(1);
            });
        });

        describe('placeholderSrc', function() {
            it('Should apply default placeholder src with blank src', function() {
                const wrapper = mount(<ZoomableImage width={allProps.width} />);

                const image = wrapper.find(Image);
                const imageURL = image.props().source.uri;

                expect(imageURL).toBe(NO_CAR_IMAGE);
            });

            it('Should apply given placeholder to Image component when src is not passed/working', function() {
                const wrapper = mount(
                    <ZoomableImage
                        width={allProps.width}
                        placeholderSrc={allProps.placeholderSrc}
                    />
                );

                const image = wrapper.find(Image);
                const imageURL = image.props().source.uri;

                expect(imageURL).toBe(allProps.placeholderSrc);
            });
        });

        describe('defaultSrc', function() {
            it('should apply given defaultSrc to Image component', function() {
                const wrapper = shallow(
                    <ZoomableImage
                        width={allProps.width}
                        defaultSrc={allProps.defaultSrc}
                    />
                );
                const image = wrapper.find(Image);
                expect(image.prop('defaultSource')).toEqual(
                    allProps.defaultSrc
                );
            });
        });

        describe('src', function() {
            it('Should apply no car image with blank src', function() {
                const wrapper = mount(<ZoomableImage width={allProps.width} />);

                const image = wrapper.find(Image);
                const imageURL = image.props().source.uri;

                expect(imageURL).toBe(NO_CAR_IMAGE);
            });

            it('Should apply given src to Image component', function() {
                const wrapper = mount(
                    <ZoomableImage width={allProps.width} src={allProps.src} />
                );

                const image = wrapper.find(Image);
                const imageURL = image.props().source.uri;

                expect(imageURL).toBe(allProps.src);
            });
        });

        describe('style', function() {
            it('Should not apply style when with blank style', function() {
                const wrapper = mount(<ZoomableImage width={allProps.width} />);

                const style = wrapper.find(Image).props().style[1];

                expect(style).toBe(null);
            });

            it('Should apply style to image', function() {
                const wrapper = mount(
                    <ZoomableImage
                        width={allProps.width}
                        style={allProps.style}
                    />
                );

                const style = wrapper.find(Image).props().style[1];

                expect(style).toBe(allProps.style);
            });
        });

        describe('wrapperStyle', function() {
            it('Should not apply style when with blank wrapperStyle', function() {
                const wrapper = mount(<ZoomableImage width={allProps.width} />);

                const style = wrapper.find(View).props().style[1];

                expect(style).toBe(null);
            });

            it('Should apply style to image wrapper', function() {
                const wrapper = mount(
                    <ZoomableImage
                        width={allProps.width}
                        wrapperStyle={allProps.wrapperStyle}
                    />
                );

                const style = wrapper.find(View).props().style[1];

                expect(style).toBe(allProps.wrapperStyle);
            });
        });
    });
});
