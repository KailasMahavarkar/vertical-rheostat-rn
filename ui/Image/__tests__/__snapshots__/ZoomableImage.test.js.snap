// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ZoomableImage /> render Should render with custom props 1`] = `
<ZoomableImage
  aspectRatio="21:9"
  data-testing-id="parent"
  defaultSrc=""
  doubleTapDelay={300}
  maxScale={4}
  onDoubleTap={[MockFunction]}
  onPinch={[MockFunction]}
  onTouchMove={[MockFunction]}
  onTouchRelease={[Function]}
  placeholderSrc="https://imgd.aeplcdn.com/310x174/n/cw/ec/54437/2021-wrangler-exterior-right-front-three-quarter.jpeg?q=75"
  shouldZoomOnPinch={false}
  src="https://imgd.aeplcdn.com/0x0/n/cw/ec/41523/sonet-exterior-right-front-three-quarter-108.jpeg?q=85"
  style={
    {
      "padding": 10,
    }
  }
  threshold={2}
  thresholdSwipeValue={50}
  width={360}
  wrapperStyle={
    {
      "padding": 10,
    }
  }
>
  <View
    accessibilityLabel="parent-zoomable-image-wrapper"
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "padding": 10,
        },
        {
          "height": 154.28571428571428,
          "width": 360,
        },
      ]
    }
    testID="parent-zoomable-image-wrapper"
  >
    <View
      accessibilityLabel="parent-zoomable-image-wrapper"
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "padding": 10,
          },
          {
            "height": 154.28571428571428,
            "width": 360,
          },
        ]
      }
      testID="parent-zoomable-image-wrapper"
    >
      <Image
        accessibilityLabel="parent-zoomable-image"
        defaultSource=""
        onError={[Function]}
        source={
          {
            "uri": "https://imgd.aeplcdn.com/0x0/n/cw/ec/41523/sonet-exterior-right-front-three-quarter-108.jpeg?q=85",
          }
        }
        style={
          [
            {
              "position": "absolute",
            },
            {
              "padding": 10,
            },
            {
              "height": 154.28571428571428,
              "left": 0,
              "top": 0,
              "width": 360,
            },
          ]
        }
        testID="parent-zoomable-image"
      >
        <Image
          accessibilityLabel="parent-zoomable-image"
          defaultSource=""
          onError={[Function]}
          source={
            {
              "uri": "https://imgd.aeplcdn.com/0x0/n/cw/ec/41523/sonet-exterior-right-front-three-quarter-108.jpeg?q=85",
            }
          }
          style={
            [
              {
                "position": "absolute",
              },
              {
                "padding": 10,
              },
              {
                "height": 154.28571428571428,
                "left": 0,
                "top": 0,
                "width": 360,
              },
            ]
          }
          testID="parent-zoomable-image"
        />
      </Image>
    </View>
  </View>
</ZoomableImage>
`;

exports[`<ZoomableImage /> render Should render with default props 1`] = `
<ZoomableImage
  aspectRatio="16:9"
  defaultSrc="https://imgd.aeplcdn.com/0x0/no-image/no-image.png"
  doubleTapDelay={500}
  maxScale={3}
  onDoubleTap={[Function]}
  onPinch={[Function]}
  onTouchMove={[Function]}
  onTouchRelease={[Function]}
  placeholderSrc="https://imgd.aeplcdn.com/0x0/no-image/no-image.png"
  shouldZoomOnPinch={false}
  src=""
  style={null}
  threshold={1.5}
  thresholdSwipeValue={50}
  width={360}
  wrapperStyle={null}
>
  <View
    accessibilityLabel={null}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      [
        {
          "overflow": "hidden",
        },
        null,
        {
          "height": 202.5,
          "width": 360,
        },
      ]
    }
    testID={null}
  >
    <View
      accessibilityLabel={null}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          null,
          {
            "height": 202.5,
            "width": 360,
          },
        ]
      }
      testID={null}
    >
      <Image
        accessibilityLabel={null}
        defaultSource="https://imgd.aeplcdn.com/0x0/no-image/no-image.png"
        onError={[Function]}
        source={
          {
            "uri": "https://imgd.aeplcdn.com/0x0/no-image/no-image.png",
          }
        }
        style={
          [
            {
              "position": "absolute",
            },
            null,
            {
              "height": 202.5,
              "left": 0,
              "top": 0,
              "width": 360,
            },
          ]
        }
        testID={null}
      >
        <Image
          accessibilityLabel={null}
          defaultSource="https://imgd.aeplcdn.com/0x0/no-image/no-image.png"
          onError={[Function]}
          source={
            {
              "uri": "https://imgd.aeplcdn.com/0x0/no-image/no-image.png",
            }
          }
          style={
            [
              {
                "position": "absolute",
              },
              null,
              {
                "height": 202.5,
                "left": 0,
                "top": 0,
                "width": 360,
              },
            ]
          }
          testID={null}
        />
      </Image>
    </View>
  </View>
</ZoomableImage>
`;
